Compilation error in file test_string.cpp with japanese locale

**Godot version:**
Reproduced on 3.1, 3.1.1, 3.1-stable and 3.1.1-stable

**OS/device including version:**
Windows 10, japanese locale, Python 3.7.3, Visual Studio Express 2019

**Issue description:**
Compilation error in file test_string.cpp

E:\godot>scons -j4 p=windows vsproj=yes
scons: Reading SConscript files ...
Configuring for Windows: target=debug, bits=default
Found MSVC compiler: amd64
Compiled program architecture will be a 64 bit executable (forcing bits=64).
YASM is necessary for WebM SIMD optimizations.
WebM SIMD optimizations are disabled. Check if your CPU architecture, CPU bits or platform are supported!
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[Initial build] [94mCompiling [95m==> [93mmain\tests\test_string.cpp[0m
[Initial build] test_string.cpp
[Initial build] main\tests\test_string.cpp(951): warning C4819: The file contains a character that cannot be represented in the current code page (932). Save the file in Unicode format to prevent data loss
[Initial build] main\tests\test_string.cpp(1060): error C2001: newline in constant
[Initial build] string.cpp(1062): error C2146: syntax error: missing ';' before identifier 'String'
[Initial build] Compiling ==> thirdparty\libvpx\vp9\common\vp9_scale.c
[Initial build] Compiling ==> thirdparty\libvpx\vp9\common\vp9_scan.c
[Initial build] scons: *** [main\tests\test_string.windows.tools.64.obj] Error 2
Compiling ==> thirdparty\libvpx\vp9\common\vp9_seg_common.c
vp9_scale.c
vp9_scan.c
vp9_seg_common.c
scons: building terminated because of errors.


**Steps to reproduce:**
Follow instructions from official docs on compilation with japanese locale

**Minimal reproduction project:**
Follow instructions from official docs on compilation with japanese locale

I added this test as part of #26760, seems like it's not that portable to define `wchar_t` literals that way, perhaps file encoding needs to be changed (`UTF-8` is by default).

@RicardRC check the file encoding for `test_string.cpp` and try resaving it with `UTF-8`/`UTF-16` encoding and see if that works.

*Note*: storing the test string with `u` prefix results in this:
```
main\tests\test_string.cpp(1071): error C2440: 'initializing': cannot convert from 'const char16_t [34]' to 'String'
```
So most likely the file encoding should be fixed instead. I don't know how git checks out files on systems with japanese locale, but something tells me it defaults to `UTF-16` which leads to this problem.

*Note*: `test_string.cpp` compiles fine with `UTF-16 LE` encoding on my system.

I reported only the error that prevents compilation. I do see another warning, on test 31, that states something related to your proposed solution.

1>main\tests\test_string.cpp(951): warning C4819: The file contains a character that cannot be represented in the current code page (932). Save the file in Unicode format to prevent data loss

UTF-8, MSVSC2019

UTF-8 with BOM results:
Just warnings on test 32, the same that you seem to have reproduced sporadically
But at least no errors, so the engine compiles and works

1>main\tests\test_string.cpp(1010): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1010): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1010): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1011): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1011): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1011): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1012): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1012): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1012): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1013): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1013): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1013): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1039): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1039): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1039): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1040): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1040): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1040): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1041): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1041): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1041): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1042): warning C4566: character represented by universal-character-name '\u00B5' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1042): warning C4566: character represented by universal-character-name '\u00BF' cannot be represented in the current code page (932)
1>main\tests\test_string.cpp(1042): warning C4566: character represented by universal-character-name '\u00FF' cannot be represented in the current code page (932)

--- a/SConstruct
+++ b/SConstruct
@@ -327,6 +327,8 @@ if selected_platform in platform_list:
         env.Append(CCFLAGS=['/EHsc'])
         if (env["werror"]):
             env.Append(CCFLAGS=['/WX'])
+        # Force to use Unicode encoding
+        env.Append(MSVC_FLAGS=['/utf8'])
     else: # Rest of the world
         shadow_local_warning = []
         all_plus_warnings = ['-Wwrite-strings']

--- a/main/tests/test_string.cpp
+++ b/main/tests/test_string.cpp
@@ -1017,8 +1017,8 @@ bool test_32() {
 	STRIP_TEST(String("abca").lstrip("a") == "bca");
 	STRIP_TEST(String("abc").rstrip("a") == "abc");
 	STRIP_TEST(String("abca").rstrip("a") == "abc");
-	// in utf-8 "¿" has the same first byte as "µ"
-	// and the same second as "ÿ"
+	// in utf-8 "¿" (\u00bf) has the same first byte as "µ" (\u00b5)
+	// and the same second as "ÿ" (\u00ff)
 	STRIP_TEST(String::utf8("¿").lstrip(String::utf8("µÿ")) == String::utf8("¿"));
 	STRIP_TEST(String::utf8("¿").rstrip(String::utf8("µÿ")) == String::utf8("¿"));
 	STRIP_TEST(String::utf8("µ¿ÿ").lstrip(String::utf8("µÿ")) == String::utf8("¿ÿ"));
@@ -1046,8 +1046,8 @@ bool test_32() {
 	STRIP_TEST(String("abca").lstrip("qwajkl") == "bca");
 	STRIP_TEST(String("abc").rstrip("qwajkl") == "abc");
 	STRIP_TEST(String("abca").rstrip("qwajkl") == "abc");
-	// in utf-8 "¿" has the same first byte as "µ"
-	// and the same second as "ÿ"
+	// in utf-8 "¿" (\u00bf) has the same first byte as "µ" (\u00b5)
+	// and the same second as "ÿ" (\u00ff)
 	STRIP_TEST(String::utf8("¿").lstrip(String::utf8("qwaµÿjkl")) == String::utf8("¿"));
 	STRIP_TEST(String::utf8("¿").rstrip(String::utf8("qwaµÿjkl")) == String::utf8("¿"));
 	STRIP_TEST(String::utf8("µ¿ÿ").lstrip(String::utf8("qwaµÿjkl")) == String::utf8("¿ÿ"));
@@ -1068,8 +1068,8 @@ bool test_33() {
 bool test_34() {
 	OS::get_singleton()->print("\n\nTest 34: Cyrillic to_lower()\n");
 
-	String upper = L"АБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯ";
-	String lower = L"абвгдеёжзийклмнопрстуфхцчшщъыьэюя";
+	String upper = String::utf8("АБВГДЕЁЖЗИЙКЛМНОПРСТУФХЦЧШЩЪЫЬЭЮЯ");
+	String lower = String::utf8("абвгдеёжзийклмнопрстуфхцчшщъыьэюя");
 
 	String test = upper.to_lower();
 

